pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -2328,3 +2501,152 @@ func TestIsBeforeFirstTaskRun_WithStartedTask(t *testing.T) {
+		finalTasks:         []v1beta1.PipelineTask{pts[0]},
+		expectedFinalTasks: []*ResolvedPipelineRunTask{noneStartedState[0]},
+	}, {
+		// tasks: [ mytask1, mytask6 with condition]
+		// finally: [mytask2]
+		name:               "DAG task (mytask1) finished, mytask6 condition failed - schedule final tasks (mytask2)",
+		state:              append(conditionCheckFailedWithOthersPassedState, noneStartedState[1]),
+		DAGTasks:           []v1beta1.PipelineTask{pts[5], pts[0]},
+		finalTasks:         []v1beta1.PipelineTask{pts[1]},
+		expectedFinalTasks: []*ResolvedPipelineRunTask{noneStartedState[1]},
+	}, {
+		// tasks: [ mytask1, mytask6 with condition]
+		// finally: [mytask2]
+		name:               "DAG task (mytask1) failed, mytask6 condition failed - schedule final tasks (mytask2)",
+		state:              append(conditionCheckFailedWithOthersFailedState, noneStartedState[1]),
+		DAGTasks:           []v1beta1.PipelineTask{pts[5], pts[0]},
+		finalTasks:         []v1beta1.PipelineTask{pts[1]},
+		expectedFinalTasks: []*ResolvedPipelineRunTask{noneStartedState[1]},
+	}, {
+		// tasks: [ mytask6 with condition, mytask7 runAfter mytask6]
+		// finally: [mytask2]
+		name:               "DAG task (mytask6) condition failed, mytask6 and mytask7 skipped - schedule final tasks (mytask2)",
+		state:              append(taskWithParentSkippedState, noneStartedState[1]),
+		DAGTasks:           []v1beta1.PipelineTask{pts[5], pts[6]},
+		finalTasks:         []v1beta1.PipelineTask{pts[1]},
+		expectedFinalTasks: []*ResolvedPipelineRunTask{noneStartedState[1]},
+	}, {
+		// tasks: [ mytask1, mytask6 with condition, mytask8 runAfter mytask6]
+		// finally: [mytask2]
+		name:               "DAG task (mytask1) finished - DAG task (mytask6) condition failed, mytask6 and mytask8 skipped - schedule final tasks (mytask2)",
+		state:              append(taskWithMultipleParentsSkippedState, noneStartedState[1]),
+		DAGTasks:           []v1beta1.PipelineTask{pts[0], pts[5], pts[7]},
+		finalTasks:         []v1beta1.PipelineTask{pts[1]},
+		expectedFinalTasks: []*ResolvedPipelineRunTask{noneStartedState[1]},
+	}, {
+		// tasks: [ mytask1, mytask6 with condition, mytask8 runAfter mytask6, mytask9 runAfter mytask1 and mytask6]
+		// finally: [mytask2]
+		name: "DAG task (mytask1) finished - DAG task (mytask6) condition failed, mytask6, mytask8, and mytask9 skipped" +
+			"- schedule final tasks (mytask2)",
+		state:              append(taskWithGrandParentSkippedState, noneStartedState[1]),
+		DAGTasks:           []v1beta1.PipelineTask{pts[0], pts[5], pts[7], pts[8]},
+		finalTasks:         []v1beta1.PipelineTask{pts[1]},
+		expectedFinalTasks: []*ResolvedPipelineRunTask{noneStartedState[1]},
+	}, {
+		//tasks: [ mytask1, mytask6 with condition, mytask8 runAfter mytask6, mytask9 runAfter mytask1 and mytask6]
+		//finally: [mytask2]
+		name: "DAG task (mytask1) finished - DAG task (mytask6) failed - mytask8 and mytask9 skipped" +
```

Comment:
Long names are informative, but a bit hard to recognise / read:
```
(dlv) p tc.name
"one dag task started executing but not finished - no final tasks...+19 more"
```

Reply:
see below, added `desc` and have `name` as a title